cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs

```diff
@@ -0,0 +1,222 @@
+      drepVkeyFp n = tempAbsPath' </> "drep-keys" </> ("drep" <> show n) </> "drep.vkey"
+
+      drepSKeyFp :: Int -> FilePath
+      drepSKeyFp n = tempAbsPath' </> "drep-keys" </> ("drep" <> show n) </> "drep.skey"
+
+  gov <- H.createDirectoryIfMissing $ tempAbsPath' </> "governance"
+
+  -- Create DRep and register them
+  let drepCertFile :: Int -> FilePath
+      drepCertFile n = gov </> "drep-keys" <> "drep" <> show n <> ".regcert"
+  forM_ [1..3] $ \n -> do
+    H.execCli' execConfig
+       [ "conway", "governance", "drep", "registration-certificate"
+       , "--drep-verification-key-file", drepVkeyFp n
+       , "--key-reg-deposit-amt", "0"
+       , "--out-file", drepCertFile n
+       ]
+
+  -- Retrieve UTxOs for registration submission
+  txin1 <- queryAnyUtxo . paymentKeyInfoAddr $ head wallets
+
+  drepRegTxbodyFp <- H.note $ work </> "drep.registration.txbody"
+  drepRegTxSignedFp <- H.note $ work </> "drep.registration.tx"
+
+  void $ H.execCli' execConfig
+    [ "conway", "transaction", "build"
+    , "--change-address", Text.unpack $ paymentKeyInfoAddr $ head wallets
+    , "--tx-in", Text.unpack $ renderTxIn txin1
+    , "--tx-out", Text.unpack (paymentKeyInfoAddr (wallets !! 1)) <> "+" <> show @Int 5_000_000
+    , "--certificate-file", drepCertFile 1
+    , "--certificate-file", drepCertFile 2
+    , "--certificate-file", drepCertFile 3
+    , "--witness-override", show @Int 4
+    , "--out-file", drepRegTxbodyFp
+    ]
+
+  void $ H.execCli' execConfig
+    [ "conway", "transaction", "sign"
+    , "--tx-body-file", drepRegTxbodyFp
+    , "--signing-key-file", paymentSKey $ paymentKeyInfoPair $ head wallets
+    , "--signing-key-file", drepSKeyFp 1
+    , "--signing-key-file", drepSKeyFp 2
+    , "--signing-key-file", drepSKeyFp 3
+    , "--out-file", drepRegTxSignedFp
+    ]
+
+  void $ H.execCli' execConfig
+    [ "conway", "transaction", "submit"
+    , "--tx-file", drepRegTxSignedFp
+    ]
+  -- Finished creating DReps
+
+  -- TODO: we could wait less: waiting 1 block should suffice.
+  DRep.waitDRepsNumber (Api.File configurationFile) (Api.File socketPath) execConfig 3
+
+  -- protocol-parameters to stdout
+  protocolParametersOut <- H.execCli' execConfig
+                             [ "conway", "query", "protocol-parameters" ]
+  H.diffVsGoldenFile
+    protocolParametersOut
+    "test/cardano-testnet-test/files/golden/queries/protocolParametersOut.txt"
+  -- protocol-parameters to a file
+  let protocolParametersOutFile = work </> "protocol-parameters-out.json"
+  H.noteM_ $ H.execCli' execConfig [ "conway", "query", "protocol-parameters"
+                                   , "--out-file", protocolParametersOutFile]
+  H.diffFileVsGoldenFile
+    protocolParametersOutFile
+    "test/cardano-testnet-test/files/golden/queries/protocolParametersFileOut.json"
+
+  -- tip to stdout
+  _ :: QueryTipLocalStateOutput <- H.noteShowM $ H.execCliStdoutToJson execConfig [ "conway", "query", "tip" ]
```

Comment:
If I'm not mistaken, the tip is also obtainable from epoch state. It may be cleaner to get it from there if possible.

Reply:
@carbolymer> I'm not sure what you mean. Do you mean calling `query ledger-state`?

If yes, I would keep the current test, as I will anyway add a test for `query ledger-state` in the future. After all, under the hood, they could be implemented differently.

Reply:
Ah nevermind. You're just testing the CLI query, and not just retrieving the tip. It's fine. 😃